src/NuGet.Core/NuGet.Packaging/PackageExtraction/StreamExtensions.cs

```diff
@@ -2,12 +2,19 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.IO;
+using System.IO.MemoryMappedFiles;
 
 namespace NuGet.Packaging
 {
     public static class StreamExtensions
     {
-        public static string CopyToFile(this Stream inputStream, string fileFullPath)
+
+        /**
+        Only files smaller than this value will be mmap'ed
+        */
+        private const long MAX_MMAP_SIZE = 10 * 1024 * 1024;
+        public static string CopyToFile(this Stream inputStream, string fileFullPath) => CopyToFile(inputStream, fileFullPath, null);
+        public static string CopyToFile(this Stream inputStream, string fileFullPath, long? size)
```

Comment:
I feel that adding `size` is just a way of allowing call sites to opt-in to this new behavior; however, isn't `size` always just `inputStream.Length`?

`Stream.Length` can throw if it's not supported, but you can handle that:
```C#
long? size = null;

if (inputStream.CanSeek)
{
    size = inputStream.Length;
}
```

If you need to keep the ability for call sites to control the MMF behavior, then you can create an `internal` extension method for this behavior (e.g.:  `CopyToFileWithPossibleMemoryMappedFile(...)`) and call it wherever you need.

Reply:
size is not available for network streams, no (I checked)